eng/build.sh

```diff
@@ -306,7 +306,7 @@ function BuildSolution {
     /p:ContinuousIntegrationBuild=$ci \
     /p:TreatWarningsAsErrors=true \
     /p:TestRuntimeAdditionalArguments=$test_runtime_args \
-    /p:DotNetBuildFromSource=$source_build \
+    /p:ArcadeBuildFromSource=$source_build \
```

Comment:
Does this cause arcade to set the `DotNetBuildFromSource` environment variable?

Reply:
Yes it does.

Reply:
`DotNetBuildFromSource` is still the appropriate variable to use when conditionalizing for source-build.